### ISSUE TITLE
Obsolete old HTTP APIs

### ISSUE THREAD

**Issue by scalablecory** [dotnet/runtime#33125]
`ObsoleteAttribute` will be getting an update to support [diagnostic IDs and URLs](https://github.com/dotnet/runtime/issues/33089) to help IDEs better manage obsolete API usage.

We should consider annotating `WebRequest` and `WebClient` with this. I frequently see asks for help with HTTP in .NET, and it's not uncommon to find them using one of these APIs without knowing we prefer them use `HttpClient` instead.

Something like

```c#
    [Obsolete(DiagnosticId="BCL1234", UrlFormat="https://aka.ms/obsoletedotnethttp")]
    public class WebRequest // Also: HttpWebRequest, FtpWebRequest, FileWebRequest (aka *WebRequest)
    {
    }

    [Obsolete(DiagnosticId="BCL1234", UrlFormat="https://aka.ms/obsoletedotnethttp")]
    public class WebClient
    {
    }
```

I would not want to create excessive warnings for users porting Framework apps to .NET 5, so this is predicated on IDE/compiler being updated to allow ignoring specific diagnostic IDs application-wide.

- [x] #41400 - Runtime obsoletion
- [x] dotnet/dotnet-api-docs#4731
- [x] dotnet/docs#20310
- [x] Redirect https://aka.ms/dotnet-warnings/SYSLIB0014 link to: ~~https://docs.microsoft.com/en-us/dotnet/core/compatibility/fx-core#webrequest-is-not-recomended-for-new-development~~ (a dedicated SYSLIB documentation page was created instead)

**Comment by karelz** [comment]
@scalablecory can you please provide guidance / options here? (the latest status of this feature)

**Comment by jeffhandley** [comment]
I left some feedback on https://github.com/dotnet/runtime/pull/41526 that likely applies to all of the obsoletions here.  I'll take a look at the other PRs though.

**Comment by karelz** [comment]
Fixed in 6.0 in PR #41400 

Note: Porting to 5.0 was rejected this late in the game -- see #41526

**Comment by aik-jahoda** [comment]
Reopening to help track the documentation.

**Comment by jeffhandley** [comment]
When this is worked on to track the documentation, please also see https://github.com/dotnet/runtime/issues/41759, where I provided some other feedback.

**Comment by gewarren** [comment]
Should this have been closed by #4731? The literal comment is `partially fixes #33125`. The dotnet/docs PR is still in draft, and the aka.ms link wasn't updated. However, with https://github.com/dotnet/docs/pull/23915, perhaps it should be redirected to the new SYSLIB0014 article instead.

**Comment by jeffhandley** [comment]
Probably not -- that might have been a false positive auto-close because of `fixes #33125`.

I do think this will get wrapped up with dotnet/docs#23915 though, and when the `SYSLIB0014` page is published, we will need to update the aka.ms link. I'll reopen this to make sure it gets tracked.

**Comment by jeffhandley** [comment]
The documentation page is in place and the aka.ms link is configured to point to it.
